packages/api/lambdas/reports/orca-backup-reconciliation-report.js

```diff
@@ -0,0 +1,374 @@
+  );
+  let nextEsItem = await esCollectionsIterator.shift();
+  while (nextEsItem) {
+    const collectionId = constructCollectionId(nextEsItem.name, nextEsItem.version);
+    const excludeFileTypes = get(nextEsItem, 'meta.excludeFileTypes');
+    if (excludeFileTypes) set(collectionsConfig, `${collectionId}.orca.excludeFileTypes`, excludeFileTypes);
+    nextEsItem = await esCollectionsIterator.shift(); // eslint-disable-line no-await-in-loop
+  }
+
+  return collectionsConfig;
+}
+
+/**
+ * check if a file should be excluded from orca backup
+ *
+ * @param {Object} collectionsConfig - collections configuration
+ * @param {Object} collectionId - collection id for the file
+ * @param {string} fileName - file name
+ * @returns {boolean} - whether the file should be excluded
+ */
+function shouldFileExcludedFromOrca(collectionsConfig, collectionId, fileName) {
+  const excludeFileTypes = get(collectionsConfig, `${collectionId}.orca.excludeFileTypes`, []);
+  if (excludeFileTypes.find((type) => fileName.endsWith(type))) return true;
+  return false;
+}
+
+/**
+ * compare cumulus granule with its orcaGranule if any, and generate report
+ *
+ * @param {Object} params
+ * @param {Object} params.collectionsConfig - collections configuration
+ * @param {Object} params.cumulusGranule - cumulus granule
+ * @param {Object} params.orcaGranule - orca granule
+ * @returns {Object} - discrepency report of the granule
+ */
+function getReportForOneGranule({ collectionsConfig, cumulusGranule, orcaGranule }) {
+  const granuleReport = {
+    ok: false,
+    okFilesCount: 0,
+    ...pick(cumulusGranule, granuleFields),
+    conflictFiles: [],
+  };
+
+  // get allFileNames(all file names) from cumulus and orca granules
+  // for each file in allFileNames:
+  //   file is in both cumulus and orca,
+  //     file should be excludedFromOrca -> add to conflictFiles with reason shouldExcludedFromOrca
+  //     file should not be excludedFromOrca -> increase okFilesCount
+  //   file is only in cumulus,
+  //     file should be excludedFromOrca -> increase okFilesCount
+  //     file type is not excludedFromOrca -> add to conflictFiles with reason 'onlyInCumulus'
+  //   file is only in orca, add file to conflictFiles with reason 'onlyInOrca'
+  // if no granule file conflicts, set granuleReport.ok to true
+
+  // reducer, key: fileName, value: file object with selected fields
+  const fileFields = ['bucket', 'key', 'cumulusArchiveLocation', 'orcaArchiveLocation', 'keyPath'];
+  const fileReducer = (accumulator, currentValue) => {
+    const fileName = currentValue.fileName
+      || path.basename(currentValue.key || currentValue.keyPath);
+    return ({
+      ...accumulator,
+      [fileName]: pick(currentValue, fileFields),
+    });
+  };
+
+  const cumulusFiles = get(cumulusGranule, 'files', []).reduce(fileReducer, {});
+  const orcaFiles = get(orcaGranule, 'files', []).reduce(fileReducer, {});
```

Comment:
So we have a single file reducer that is used for both cumulus & ORCA files? But I assume properties like `orcaArchiveLocation` only exist on ORCA files and `bucket`/`key` only exist on Cumulus files?

I think it might be clearer to have separate reducers for the ORCA and Cumulus files